libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/infopanel/InfoPanelHeaderComponent.kt

```diff
@@ -4,33 +4,38 @@ import androidx.core.view.isVisible
 import com.mapbox.navigation.core.MapboxNavigation
 import com.mapbox.navigation.dropin.DropInNavigationViewContext
 import com.mapbox.navigation.dropin.R
-import com.mapbox.navigation.dropin.component.destination.DestinationState
-import com.mapbox.navigation.dropin.component.navigationstate.NavigationState
+import com.mapbox.navigation.dropin.component.destination.DestinationAction
+import com.mapbox.navigation.dropin.component.navigation.NavigationState
+import com.mapbox.navigation.dropin.component.navigation.NavigationStateAction
 import com.mapbox.navigation.dropin.component.routefetch.RoutesAction
+import com.mapbox.navigation.dropin.component.routefetch.RoutesState
 import com.mapbox.navigation.dropin.databinding.MapboxInfoPanelHeaderLayoutBinding
 import com.mapbox.navigation.dropin.lifecycle.UIComponent
-import kotlinx.coroutines.flow.StateFlow
+import com.mapbox.navigation.utils.internal.logW
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.takeWhile
+import kotlinx.coroutines.isActive
+import kotlinx.coroutines.launch
 
 internal class InfoPanelHeaderComponent(
     private val binding: MapboxInfoPanelHeaderLayoutBinding,
-    private val context: DropInNavigationViewContext
+    context: DropInNavigationViewContext
```

Comment:
Same as above. Can we inject the viewmodels?

Reply:
I keep going back and forth on this honestly.. 

On one hand, it's what we want to do long term. On the other hand, it requires a lot more test code

Reply:
Right now the `InfoPanelHeaderComponent` is using every view model. The plan is also to break this this apart into smaller components.

The smaller components are depending on individual view models, but this one is still responsible for everything

Reply:
moving them